frontend/src/lib/components/Cards/TextCard/TextCard.tsx

```diff
@@ -0,0 +1,289 @@
+import clsx from 'clsx'
+import { DashboardTile, DashboardType, InsightColor, Tileable } from '~/types'
+import { LemonButton, LemonButtonWithPopup, LemonDivider, LemonModal, LemonTextArea } from '@posthog/lemon-ui'
+import { More } from 'lib/components/LemonButton/More'
+import { useActions, useValues } from 'kea'
+import { Field, Form } from 'kea-forms'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+import { Splotch, SplotchColor } from 'lib/components/icons/Splotch'
+import { capitalizeFirstLetter } from 'lib/utils'
+import ReactMarkdown from 'react-markdown'
+import { IconMarkdown } from 'lib/components/icons'
+import { Tabs } from 'antd'
+import { UserActivityIndicator } from 'lib/components/UserActivityIndicator/UserActivityIndicator'
+import { textCardModalLogic } from 'lib/components/Cards/TextCard/textCardModalLogic'
+import { dashboardsModel } from '~/models/dashboardsModel'
+
+export function TextTileModal({
+    isOpen,
+    onClose,
+    dashboard,
+    textTileId,
+}: {
+    isOpen: boolean
+    onClose: () => void
+    dashboard: DashboardType
+    textTileId: number | 'new' | null
+}): JSX.Element {
+    const modalLogic = textCardModalLogic({ dashboard, textTileId: textTileId ?? 'new', onClose })
+    const { isTextTileSubmitting } = useValues(modalLogic)
+    const { submitTextTile, resetTextTile } = useActions(modalLogic)
+
+    const handleClose = (): void => {
+        resetTextTile()
+        onClose()
+    }
+
+    return (
+        <LemonModal
+            closable={true}
+            isOpen={isOpen}
+            title={''}
+            onClose={handleClose}
+            footer={
+                <>
+                    <LemonButton disabled={isTextTileSubmitting} type="secondary" onClick={handleClose}>
+                        Cancel
+                    </LemonButton>
+                    <LemonButton
+                        disabled={isTextTileSubmitting}
+                        loading={isTextTileSubmitting}
+                        form="text-tile-form"
+                        htmlType="submit"
+                        type="primary"
+                        onClick={submitTextTile}
+                        data-attr={textTileId === 'new' ? 'save-new-text-tile' : 'edit-text-tile-text'}
+                    >
+                        Save
+                    </LemonButton>
+                </>
+            }
+        >
+            <Form
+                logic={textCardModalLogic}
+                props={{ dashboard: dashboard, textTileId: textTileId }}
+                formKey={'textTile'}
+                id="text-tile-form"
+                className=""
+                enableFormOnSubmit
+            >
+                <Field name="body" label="">
+                    {({ value, onChange }) => (
+                        <Tabs>
+                            <Tabs.TabPane tab="Write" key="write-card" destroyInactiveTabPane={true}>
```

Comment:
This is great – can this be extracted into its own component (something like `LemonTextArea.Markdown`)? Not even as a DRY thing at this point, but it'd be more readable than `{({ value, onChange }) => ...` and generally it'd be really useful if we enabled using Markdown in the app more (e.g. in various descriptions).

Reply:
Very quickly pulled this out. Can punt down the line or pair on polishing it ...

Reply:
Looks good!